File: backend/utils/data_cleaner.py
import re

def clean_price(value):
    if not value:
        return None
    value = re.sub(r"[^\d.]", "", value)
    return float(value) if value else None

def clean_discount(value):
    if not value:
        return None
    match = re.search(r"\d+", value)
    return int(match.group()) if match else None

File: node_backend/utils/dataCleaner.js
/**
 * Data cleaning utilities — equivalent to utils/data_cleaner.py
 */

function cleanPrice(value) {
  if (!value) return null;
  const cleaned = String(value).replace(/[^\d.]/g, '');
  if (!cleaned) return null;
  const num = parseFloat(cleaned);
  return isNaN(num) ? null : num;
}

function cleanDiscount(value) {
  if (!value) return null;
  const match = String(value).match(/\d+/);
  return match ? parseInt(match[0], 10) : null;
}

module.exports = { cleanPrice, cleanDiscount };
